### PR TITLE
59 frontend apply quiz result follow UI

### PR DESCRIPTION
**Pull request by sarunnut1236**
Screenshot 2024-07-26 020429
screencapture-localhost-4200-test-result-2024-07-26-02_04_33

### CODE REVIEW

File: frontend/src/app/features/quiz/result/result-character/result-character.component.ts [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+
+  public bgLink = '';
+  public characterNameEn = '';
+  public characterNameTh = '';
+  public characterTitle = '';
+  public characterImgLink = '';
+  public characterDescription = '';
+  public characterAttributes: string[] = [];
+  public attributesChipColor = '';
+  public shinningMethod = '';
```
Comment (viruskizz) [inline frontend/src/app/features/quiz/result/result-character/result-character.component.ts:31]: Saggestion: Use as object in stead of single var ?

File: frontend/src/app/features/quiz/result/result-character/result-character.component.ts [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+        ];
+        this.attributesChipColor = 'bg-[#50eac6]';
+        this.shinningMethod =
+          '“ส่องแสงของเธอด้วยการส่งต่อความช่วยเหลือของเธอไปยังทุกคนที่เธอสามารถช่วยได้! เริ่มจากแบ่งขนมอร่อยๆ ให้คุณน้าข้างบ้านที่ชอบมาเม้ากับแม่เราดูสิ!  ᕕ( ᐛ )ᕗ ”';
+        break;
+      default:
+        break;
+    }
+  }
+}
```
Comment (viruskizz) [inline frontend/src/app/features/quiz/result/result-character/result-character.component.ts:161]: Saggestion: mock as array and user array.find be better

File: frontend/src/app/features/quiz/result/result.component.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+      case 1:
+        if (this.hasResult) {
+          this.isClickedToShowResult = true;
+          this.displayIndex++;
+        }
+        break;
+      default:
+        break;
+    }
+  }
```
Comment (viruskizz) [inline frontend/src/app/features/quiz/result/result.component.ts:42]: Saggestion: if condition in switch case ?
Why don't we use and condition ? It shorter and nice

File: frontend/src/app/shared/background/background.component.html [hunk 4]
```diff
@@ -0,0 +1,2 @@
+<img *ngIf="isScrolling" [src]="src" alt="background" class="w-full min-h-full max-h-fit absolute -z-10">
```
Comment (viruskizz) [inline frontend/src/app/shared/background/background.component.html:1]: Still we use this file ?

File: frontend/src/app/features/quiz/result/result-character/character-attributes/character-attributes.component.ts [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+@Component({
+  selector: 'app-character-attributes',
+  standalone: true,
+  imports: [SharedModule, CommonModule],
+  templateUrl: './character-attributes.component.html',
+  styleUrl: './character-attributes.component.css',
+})
+export class CharacterAttributesComponent implements OnInit {
+  @Input() public characterAttributes!: string[];
+  @Input() public color!: string;
```
Comment (hazamashoken) [inline frontend/src/app/features/quiz/result/result-character/character-attributes/character-attributes.component.ts:14]: Don't bang the variable, 
`@Input({require: true})`

File: frontend/src/app/features/quiz/result/result-character/result-character.component.ts [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+    SharedModule,
+    CommonModule,
+    SvgIconComponent,
+    CharacterAttributesComponent,
+  ],
+  templateUrl: './result-character.component.html',
+  styleUrl: './result-character.component.css',
+})
+export class ResultCharacterComponent {
+  @Input() public character!: Character;
```
Comment (hazamashoken) [inline frontend/src/app/features/quiz/result/result-character/result-character.component.ts:21]: Same

File: frontend/src/app/features/test/result-test/result-test.component.html [hunk 7]
```diff
@@ -1,54 +1,3 @@
...
-        <app-button>Buy ticket</app-button>
-        <div class="flex flex-row justify-center items-center">    
-            <img src="images/goodgeeklogo-256.png" alt="GoodGeek Club" width="100px">
-            <p>YNS</p>
-            <p>Sponsor</p>
-        </div>
-    </section>
-</ng-template>
\ No newline at end of file
+<div class="w-[434px] h-[941px]">
```
Comment (hazamashoken) [inline frontend/src/app/features/test/result-test/result-test.component.html:1]: Becareful

**Review by hazamashoken — CHANGES_REQUESTED** [review]

File: frontend/src/app/features/quiz/result/result-character/result-character.component.ts (hunk 2 above)
Comment (sarunnut1236) [reply frontend/src/app/features/quiz/result/result-character/result-character.component.ts:161]: can i use .filter() ??

File: frontend/src/app/features/quiz/result/result.component.ts (hunk 3 above)
Comment (sarunnut1236) [reply frontend/src/app/features/quiz/result/result.component.ts:42]: in my opinion, this code is more undderstandable for me

File: frontend/src/app/features/quiz/result/result-character/result-character.component.ts (hunk 1 above)
Comment (sarunnut1236) [reply frontend/src/app/features/quiz/result/result-character/result-character.component.ts]: Done

File: frontend/src/app/features/quiz/result/result-character/result-character.component.ts (hunk 2 above)
Comment (sarunnut1236) [reply frontend/src/app/features/quiz/result/result-character/result-character.component.ts]: Done

File: frontend/src/app/features/quiz/result/result.component.ts (hunk 3 above)
Comment (sarunnut1236) [reply frontend/src/app/features/quiz/result/result.component.ts]: Done

File: frontend/src/app/features/quiz/result/result-character/character-attributes/character-attributes.component.ts (hunk 5 above)
Comment (sarunnut1236) [reply frontend/src/app/features/quiz/result/result-character/character-attributes/character-attributes.component.ts]: Done

File: frontend/src/app/features/quiz/result/result-character/result-character.component.ts (hunk 6 above)
Comment (sarunnut1236) [reply frontend/src/app/features/quiz/result/result-character/result-character.component.ts]: Done

File: frontend/src/app/features/test/result-test/result-test.component.html (hunk 7 above)
Comment (sarunnut1236) [reply frontend/src/app/features/test/result-test/result-test.component.html]: Deleted

**Review by viruskizz — APPROVED** [review]
GG

**Review by hazamashoken — APPROVED** [review]

**Pull request merged**